File: 121. Best Time to Buy and Sell Stock/main.py
"""
Author: Amal Matty Antony 
"""
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        min_so_far = prices[0]
        for i in range(1,len(prices)):
            profit = max(profit, prices[i]-min_so_far)
            min_so_far = min(min_so_far, prices[i])
        return profit
        

File: 121. Best Time to Buy and Sell Stock/main.cpp
// Solution By Vandit Jain (jainvandit99)
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // Maintain Max and min and keep iterating
        int max = 0;
        int min = INT_MAX;
        for(int i=0;i<prices.size();i++){
            if(prices[i] - min > max){
                max = prices[i] - min;
            }
            if(prices[i] < min){
                min = prices[i];
            }
        }
        
        return max;
    }
};